kafka/example_test.go

```diff
@@ -21,26 +20,37 @@ func ExamplePublisher() {
 }
 
 func ExampleSubscriber() {
-	ctx := context.TODO()
-	sub, err := bps.NewSubscriber(ctx, "kafka://10.0.0.1:9092,10.0.0.2:9092,10.0.0.3:9092/?client.id=my-client&kafka.version=2.3.0&channel.buffer.size=1024")
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	subscriber, err := bps.NewSubscriber(ctx, "kafka://10.0.0.1:9092,10.0.0.2:9092,10.0.0.3:9092/?client.id=my-client&kafka.version=2.3.0&channel.buffer.size=1024")
 	if err != nil {
 		panic(err.Error())
 	}
-	defer sub.Close()
-
-	// can be just context.WithCancel - `cancel` can be used to stop consuming:
-	ctx, cancel := context.WithTimeout(ctx, time.Second)
-	defer cancel()
+	defer subscriber.Close()
 
-	// will block till context is cancelled:
-	err = sub.Topic("topic").Subscribe(
-		ctx,
+	var subscription bps.Subscription
+	subscription, err = subscriber.Topic("topic").Subscribe(
 		bps.HandlerFunc(func(msg bps.SubMessage) {
 			_, _ = fmt.Printf("%s\n", string(msg.Data()))
+
+			// recipe to abort everything on irrecoverable handler errors:
+			_ = subscription.Close() // abort subscription, so current message won't be Ack-ed
+			cancel()                 // and cancel parent/program context to finally exit
```

Comment:
this is a slightly confusing example, I find. either you use `cancel()` OR `_ = subscription.Close()`, both should be doing the same thing, i.e. exit the handler. I am not a fan of that `_ = subscription.Close()` inside the handler, it's really making it hard to understand/follow. We are not bothered with any acking at the moment, so why mention it at all?